Python/dynload_win.c

```diff
@@ -240,8 +240,8 @@ dl_funcptr _PyImport_FindSharedFuncptrWindows(const char *prefix,
                This should not happen if called correctly. */
             if (theLength == 0) {
                 message = PyUnicode_FromFormat(
-                    "DLL load failed with error code %u",
-                    errorCode);
+                    "While importing %s, DLL load failed with error code %u",
+                    shortname, errorCode);
```

Comment:
You could append the extension module short name on the end of the existing message, i.e. "DLL load failed with error code %u while importing %s". This keeps it more consistent with the previous wording.

This will be helpful to novices.  That said, with existing support, the `name` and `path` are accessible as attributes of the `ImportError`. For example, after temporarily renaming DLLs/sqlite3.dll:

```
>>> try: import sqlite3
... except ImportError as e: print(f'name: {e.name}\npath: {e.path}')
...
name: _sqlite3
path: C:\Program Files\Python37\DLLs\_sqlite3.pyd
```

Reply:
I updated the error message based on your suggestion.